### PR TITLE
Add non word characters to php that excludes the dollar sign from a valid variable name

### PR DESCRIPTION
**Pull request by mostafaeweda**
@kevinsawicki @bolinfest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "language-php",
-  "version": "0.23.0",
+  "version": "0.24.0",
```
Comment (kevinsawicki) [inline package.json]: This gets changed as part of `apm publish`, please revert this change.

Comment (mostafaeweda) [inline package.json]: Oh, OK

**Comment by kevinsawicki** [comment]
Thanks!
